server/src/main/java/com/linecorp/centraldogma/server/CentralDogma.java

```diff
@@ -326,6 +326,17 @@ public CompletableFuture<Void> start() {
      */
     public CompletableFuture<Void> stop() {
         serverHealth.setHealthy(false);
+
+        final Optional<GracefulShutdownTimeout> gracefulTimeoutOpt = cfg.gracefulShutdownTimeout();
+        if (gracefulTimeoutOpt.isPresent()) {
+            try {
+                // Sleep so that clients have some time to redirect traffic according to the health status
+                Thread.sleep(gracefulTimeoutOpt.get().quietPeriodMillis());
+            } catch (InterruptedException e) {
+                logger.debug("Interrupted while waiting for quiet period", e);
```

Comment:
Should we restore the interruption status so that the code in the up-stack can see that an interrupt was issued?
```suggestion
                logger.debug("Interrupted while waiting for quiet period", e);
                Thread.currentThread().interrupt();
```